src/legacy/ui/public/visualize/loader/pipeline_helpers/build_pipeline.ts

```diff
@@ -259,41 +260,41 @@ const buildVisConfig: BuildVisConfigFunction = {
     };
     return visConfig;
   },
-  metric: (visState, schemas) => {
-    const visConfig = visState.params;
-    visConfig.metric.metrics = schemas.metric;
+  metric: schemas => {
+    const visConfig = { dimensions: {} } as any;
+    visConfig.dimensions.metrics = schemas.metric;
```

Comment:
Why is this put into dimensions now? Was this a bug in the old version? I realize this is the new pipeline infrastructure, just asking to learn about the code base.

Reply:
historically metric had a bit different structure of params that other visualizations (everything nested under `metric`). Initially i tried to follow that but realized its causing additional problems as then we have to check for that special condition in various places.